src/views/swap/swapReducer.js

```diff
@@ -49,7 +53,10 @@ const reducer = (state = initalState, action) => {
         ...state,
         swapInfo: {
           ...state.swapInfo,
-          invoice: action.payload,
+          invoice: {
+            ...state.swapInfo,
+            ...action.payload,
```

Comment:
Why not leave it the way it was? The invoice should just be `action.payload` shouldn't it?